code/object/collideshipweapon.cpp

```diff
@@ -490,23 +490,21 @@ static int ship_weapon_check_collision(object *ship_objp, object *weapon_objp, f
 			ship_weapon_do_hit_stuff(ship_objp, weapon_objp, &mc.hit_point_world, &mc.hit_point, quadrant_num, mc.hit_submodel, mc.hit_normal);
 		}
 
-		if (Script_system.IsActiveAction(CHA_COLLIDEWEAPON) && !(weapon_override && !ship_override))
-		{
-			Script_system.SetHookObjects(4, "Self", ship_objp, "Object", weapon_objp, "Ship", ship_objp, "Weapon", weapon_objp);
-			Script_system.SetHookVar("Hitpos", 'o', scripting::api::l_Vector.Set(mc.hit_point_world));
-			Script_system.RunCondition(CHA_COLLIDEWEAPON, ship_objp, weapon_objp);
-			Script_system.RemHookVars({ "Self", "Object", "Ship", "Weapon", "Hitpos" });
+		if (scripting::hooks::OnWeaponCollision->isActive() && !(weapon_override && !ship_override)) {
+			scripting::hooks::OnWeaponCollision->run(scripting::hooks::CollisionConditions{ {ship_objp, weapon_objp} },
+				scripting::hook_param_list(scripting::hook_param("Self", 'o', ship_objp),
+					scripting::hook_param("Object", 'o', weapon_objp),
+					scripting::hook_param("Ship", 'o', ship_objp),
+					scripting::hook_param("Weapon", 'o', weapon_objp),
+					scripting::hook_param("Hitpos", 'o', mc.hit_point_world)));
 		}
-
-		if (scripting::hooks::OnShipCollision->isActive() && ((weapon_override && !ship_override) || (!weapon_override && !ship_override)))
-		{
-			scripting::hooks::OnShipCollision->run(
+		if (scripting::hooks::OnShipCollision->isActive() && ((weapon_override && !ship_override) || (!weapon_override && !ship_override))) {
```

Comment:
It's a little strange, but I believe it to be intentional.
This works out to "The hooks to check for override are always run".
The main hooks are run, unless the other collision has been overridden, rendering the collision void. If there is a conflict and both want to override, OnWeaponCollision gets precedence, and OnShipCollision isn't run.
Yes, this condition does simplify to ``if (scripting::hooks::OnShipCollision->isActive() & !ship_override)``.
In general, it will need another round of analyzing whether the different collision checks all behave identically, but that will need more time to validate that older scripts won't be adversely affected. For now, I'll simplify the check, but keep its logic as is